assets/src/scripts/monitoring-location/iv-data-utils.js

```diff
@@ -0,0 +1,114 @@
+    const celsiusTemperatureCodesWithFahrenheitCounterparts = ['00020', '00010', '45589'];
+    // If the current parameter code is one that could match a measured Fahrenheit parameter, cross reference the
+    // parameter with the known corresponding codes that may match a value in the application state.
+    if (celsiusTemperatureCodesWithFahrenheitCounterparts.includes(parameterCode)) {
+        switch(parameterCode) {
+            case '00020':
+                allVariableParameterCodes.includes('00021')? isParameterMatching = true : isParameterMatching = false;
+                break;
+            case '00010':
+                allVariableParameterCodes.includes('00011')? isParameterMatching = true : isParameterMatching = false;
+                break;
+            case '45589':
+                allVariableParameterCodes.includes('45590')? isParameterMatching = true : isParameterMatching = false;
+                break;
+            default:
+                isParameterMatching = false;
+        }
+    } else {
+        isParameterMatching = false;
+    }
+
+    return isParameterMatching;
+};
+
+/*
+* Converts a Celsius 'variable' from the NWIS system to one we can use to show Fahrenheit.
+* @param {Object} NWISVariable - Has various properties related to descriptions of data
+* @return {Object} NWIS variable converted for use with Fahrenheit
+ */
+const createConvertedVariable = function(NWISVariable) {
+    NWISVariable.variableName = NWISVariable.variableName.replace('C', 'F (calculated)');
+    NWISVariable.variableDescription = NWISVariable.variableDescription.replace('Celsius', 'Fahrenheit (calculated)');
+    NWISVariable.unit.unitCode = NWISVariable.unit.unitCode.replace('C', 'F');
+    NWISVariable.variableCode.value = `${NWISVariable.variableCode.value}${config.CALCULATED_TEMPERATURE_VARIABLE_CODE}`;
+    NWISVariable.variableCode.variableID = `${NWISVariable.variableCode.variableID}${config.CALCULATED_TEMPERATURE_VARIABLE_CODE}`;
+    NWISVariable.oid = `${NWISVariable.oid}${config.CALCULATED_TEMPERATURE_VARIABLE_CODE}`;
+
+    return  {
+        [NWISVariable.oid]: NWISVariable
+    };
+};
+
+/*
+* Converts the temperature values (points) in the cloned time series from Celsius to Fahrenheit
+* @param {Object} timeSeries - a time series with Celsius data values
+* @param {Object} tsRequestKey - the time series request key
+* @param (Object} parameterCode - parameter code (should always be a Celsius code)
+* @return {Object} A time series with points converted from Celsius to Fahrenheit
+*   important properties (there are many others)
+*   @prop - {Object) points - An object containing temperature data
+*   @prop - {String} variable - The 'variable' from the NWIS system, will have a suffix indicating a converted value
+ */
+const createConvertedTimeSeries = function(timeSeries, tsRequestKey, parameterCode) {
+    timeSeries.points.forEach(temperaturePoint => {
```

Comment:
I see that you are passing in a cloned copy of timeSeries, but this function still mutates the contents of the parameter, timeSeries, thereby having a side effect. I'm not crazy about this but since it is an internal function I think it is okay. I'm worried that this may cause weird issues at some point. I may take a crack at refactoring this if I get done with my tickets this sprint.

Reply:
True. This is challenging set of events. I'm really interested in seeing an alternaltive path.